versioning/scripts/dockerfiles/main.go

```diff
@@ -227,33 +230,41 @@ func check(e error) {
 }
 
 func main() {
-	templateDirPtr := flag.String("template_dir", "templates", "Path to directory containing Dockerfile.template and any other files to copy over")
+	defaultTemplateDirPtr := flag.String("template_dir", "templates", "Path to directory containing Dockerfile.template and any other files to copy over")
 	verifyPtr := flag.Bool("verify_only", false, "Verify dockerfiles")
+	var version = "0.01"
+
+	// printing version of the tool; version should be bumped up
+	// whenever one makes changes (initially it'll be done manually and ultimately automatically)
+	log.Printf("dockerfiles version: %s", version)
+
 	flag.Parse()
 
 	var spec versions.Spec
 	spec = versions.LoadVersions("versions.yaml")
 
-	templatePath := filepath.Join(*templateDirPtr, "Dockerfile.template")
-	templateData, err := ioutil.ReadFile(templatePath)
-	templateString := string(templateData)
-	check(err)
+	for _, version := range spec.Versions {
+		// templatePath - path to Dockerfile.template
+		templatePath := filepath.Join(*defaultTemplateDirPtr, version.TemplateSubDir, "Dockerfile.template")
+		templateData, err := ioutil.ReadFile(templatePath)
+		templateString := string(templateData)
+		check(err)
 
-	tmpl, err := template.
-		New("dockerfileTemplate").
-		Funcs(template.FuncMap{"KeyServersRetryLoop": funcKeyServersRetryLoop}).
-		Parse(templateString)
-	check(err)
+		tmpl, err := template.
+			New("dockerfileTemplate").
+			Funcs(template.FuncMap{"KeyServersRetryLoop": funcKeyServersRetryLoop}).
+			Parse(templateString)
+		check(err)
 
-	if *verifyPtr {
-		failureCount := verifyDockerfiles(spec, *tmpl)
-		failureCount += verifyCopiedFiles(spec, *templateDirPtr)
-		os.Exit(failureCount)
-	} else {
-		for _, version := range spec.Versions {
+		if *verifyPtr {
+			failureCount := verifyDockerfiles(spec, *tmpl)
+			failureCount += verifyCopiedFiles(spec, *defaultTemplateDirPtr)
```

Comment:
This construct is strange, as `verifyCopiedFiles` loops through the versions again, and this block ends with an `os.Exit()`. Not related to this PR--would be nice to refactor this into something more sensible.